testing/ostest/ostest_main.c

```diff
@@ -347,13 +375,16 @@ static int user_main(int argc, char *argv[])
       check_test_memory_usage();
 #endif
 
+#ifndef CONFIG_BUILD_KERNEL
       /* Checkout task_restart() */
 
       printf("\nuser_main: task_restart test\n");
       restart_test();
       check_test_memory_usage();
+#endif
 
-#ifdef CONFIG_SCHED_WAITPID
+#if defined(CONFIG_SCHED_WAITPID) && \
+    (!defined(CONFIG_BUILD_KERNEL) || defined(CONFIG_SCHED_CHILD_STATUS))
```

Comment:
why check CONFIG_BUILD_KERNEL

Reply:
waitpid function is not completed, and child exit status is wrong. Should enable SCHED_CHILD_STATUS in kernel mode.